packages/internal/bridge/sdk/src/tokenBridge.ts

```diff
@@ -547,143 +477,640 @@ export class TokenBridge {
-    if (bytecode.length <= 2) return true;
+    if (bytecode.length <= 2) return;
 
+    const ABI = ['function receive()'];
     const contract = new ethers.Contract(address, ABI, provider);
 
     try {
       // try to estimate gas for the receive function, if it works it exists
       await contract.estimateGas.receive();
-      return true;
     } catch {
       try {
         // if receive fails, try to estimate this way which will work if a fallback function is present
         await provider.estimateGas({ to: address });
-        return true;
       } catch {
         // no receive or fallback
-        return false;
+        throw new BridgeError(
+          `address ${address} is not a valid receipient`,
+          BridgeErrorType.INVALID_RECIPIENT,
+        );
       }
     }
   }
 
-  private async getBridgeTx(
-    sender:string,
-    recipient:string,
-    amount:ethers.BigNumber,
-    token:string,
-    sourceChainId: string,
-    destinationChainId: string,
-    provider: ethers.providers.Provider,
-    gasMultiplier: number = 1.1,
-  ): Promise<BridgeTxResponse> {
-    const canReceive:boolean = await this.checkReceiver(provider, recipient);
+  private async validateBridgeReqArgs(
+    req: BridgeBundledTxRequest,
+  ) {
+    // Validate chain ID.
+    await this.validateChainIds(req.sourceChainId, req.destinationChainId);
 
-    if (!canReceive) {
+    // Validate address
+    if (!ethers.utils.isAddress(req.senderAddress) || !ethers.utils.isAddress(req.recipientAddress)) {
       throw new BridgeError(
-        `address ${recipient} is not a valid receipient`,
-        BridgeErrorType.INVALID_RECIPIENT,
+        `address ${req.senderAddress} or ${req.recipientAddress} is not a valid address`,
+        BridgeErrorType.INVALID_ADDRESS,
       );
     }
 
-    const getContractRes = await this.getBridgeContract(sourceChainId);
+    // Validate amount
+    if (req.amount.isNegative() || req.amount.isZero()) {
+      throw new BridgeError(
+        `deposit amount ${req.amount.toString()} is invalid`,
+        BridgeErrorType.INVALID_AMOUNT,
+      );
+    }
 
-    const fees:BridgeFeeResponse = await this.getFeePrivate({
-      action: getContractRes.contractAction,
-      gasMultiplier,
-      sourceChainId,
-      destinationChainId,
-      token,
-      amount,
-    });
+    // If the token is not native, it must be a valid address
+    if (req.token.toUpperCase() !== NATIVE && !ethers.utils.isAddress(req.token)) {
+      throw new BridgeError(
+        `token address ${req.token} is not a valid address`,
+        BridgeErrorType.INVALID_ADDRESS,
+      );
+    }
+  }
 
-    const data = await this.getTxData(
+  private async getDynamicDepositGas(
+    sourceChainId: string,
+    sender: string,
+    recipient: string,
+    token: FungibleToken,
+    amount: ethers.BigNumber,
+  ): Promise<DynamicGasEstimatesResponse> {
+    const simulations: Array<any> = [];
```

Comment:
Should we consider using stricter types here?

Reply:
Would be nice to define the simulation type..
```
{
        network_id,
        estimate_gas,
        simulation_type,
        from,
        to,
        input,
      }

Reply:
@dreamoftrees In the backlog https://immutable.atlassian.net/browse/SMR-2719

<img src="https://media3.giphy.com/media/d3mlE7uhX8KFgEmY/giphy.gif"/>